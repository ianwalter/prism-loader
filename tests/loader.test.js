const { join } = require('path')

const webpack = require('webpack')
const MemoryFileSystem = require('memory-fs')

test('loader adds syntax highlighting HTML', done => {
  const compiler = webpack({
    entry: join(__dirname, 'fixtures/test.html'),
    module: {
      rules: [
        {
          test: /\.html$/,
          use: [
            {
              loader: '@ianwalter/callback-loader',
              options: {
                callback: function (content) {
                  expect(content).toMatchSnapshot()
                  done()
                }
              }
            },
            {
              loader: join(__dirname, '../')
            }
          ]
        }
      ]
    }
  })
  compiler.outputFileSystem = new MemoryFileSystem()
  compiler.run()
})
