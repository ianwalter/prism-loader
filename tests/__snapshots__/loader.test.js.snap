// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`loader adds syntax highlighting HTML 1`] = `
"<!doctype html><html><head>
    <title>prism-loader demo</title>
    <style>
      pre {
        color: #f8f8f2;
        font-size: 1.125rem;
        background: #282a36;
      }
      .token.operator,
      .token.keyword {
        color: #ff79c6;
      }
      .token.string {
        color: #f1fa8c;
      }
      .token.comment {
        color: #6272a4;
      }
      .token.global {
        color: #bd93f9;
      }
      .token.function {
        color: #50fa7b;
      }
      .token.class-name,
      .token.function.method {
        color: #8be9fd;
      }
    </style>
  </head>
  <body>

    <pre>      <code class=\\"language-javascript\\">
        <span class=\\"token keyword\\">const</span> Sled <span class=\\"token operator\\">=</span> <span class=\\"token function\\">require</span><span class=\\"token punctuation\\">(</span><span class=\\"token string\\">'@ianwalter/sled'</span><span class=\\"token punctuation\\">)</span>
        <span class=\\"token keyword\\">const</span> sled <span class=\\"token operator\\">=</span> <span class=\\"token keyword\\">new</span> <span class=\\"token class-name\\">Sled</span><span class=\\"token punctuation\\">(</span><span class=\\"token global\\">document</span><span class=\\"token punctuation\\">.</span><span class=\\"token method function\\">querySelector</span><span class=\\"token punctuation\\">(</span><span class=\\"token string\\">'#myEl'</span><span class=\\"token punctuation\\">)</span><span class=\\"token punctuation\\">)</span>

        <span class=\\"token keyword\\">function</span> <span class=\\"token function\\">getHeight</span> <span class=\\"token punctuation\\">(</span><span class=\\"token punctuation\\">)</span> <span class=\\"token punctuation\\">{</span>
          <span class=\\"token global\\">console</span><span class=\\"token punctuation\\">.</span><span class=\\"token method function\\">log</span><span class=\\"token punctuation\\">(</span><span class=\\"token string\\">'ENTER'</span><span class=\\"token punctuation\\">)</span>
          <span class=\\"token keyword\\">return</span> <span class=\\"token string\\">'100px'</span>
        <span class=\\"token punctuation\\">}</span>

        <span class=\\"token comment\\">// Slide to a specific height:</span>
        sled<span class=\\"token punctuation\\">.</span><span class=\\"token method function\\">slide</span><span class=\\"token punctuation\\">(</span><span class=\\"token function\\">getHeight</span><span class=\\"token punctuation\\">(</span><span class=\\"token punctuation\\">)</span><span class=\\"token punctuation\\">)</span>

        <span class=\\"token comment\\">// Toggle between 0 height and the elements natural height.</span>
        sled<span class=\\"token punctuation\\">.</span><span class=\\"token method function\\">toggle</span><span class=\\"token punctuation\\">(</span><span class=\\"token punctuation\\">)</span>
      </code>
    </pre>

    <pre>      <code class=\\"language-javascript\\">
        <span class=\\"token keyword\\">import</span> Sled <span class=\\"token keyword\\">from</span> <span class=\\"token string\\">'@ianwalter/sled'</span>

        <span class=\\"token keyword\\">let</span> sled

        <span class=\\"token keyword\\">export</span> <span class=\\"token keyword\\">default</span> <span class=\\"token punctuation\\">{</span>
          <span class=\\"token function\\">mounted</span> <span class=\\"token punctuation\\">(</span><span class=\\"token punctuation\\">)</span> <span class=\\"token punctuation\\">{</span>
            sled <span class=\\"token operator\\">=</span> <span class=\\"token keyword\\">new</span> <span class=\\"token class-name\\">Sled</span><span class=\\"token punctuation\\">(</span><span class=\\"token global\\">this</span><span class=\\"token punctuation\\">.</span>$refs<span class=\\"token punctuation\\">.</span>menu<span class=\\"token punctuation\\">)</span>
          <span class=\\"token punctuation\\">}</span><span class=\\"token punctuation\\">,</span>
          methods<span class=\\"token punctuation\\">:</span> <span class=\\"token punctuation\\">{</span>
            toggleMenu<span class=\\"token punctuation\\">:</span> <span class=\\"token punctuation\\">(</span><span class=\\"token punctuation\\">)</span> <span class=\\"token operator\\">=></span> sled<span class=\\"token punctuation\\">.</span><span class=\\"token method function\\">toggle</span><span class=\\"token punctuation\\">(</span><span class=\\"token punctuation\\">)</span>
          <span class=\\"token punctuation\\">}</span>
        <span class=\\"token punctuation\\">}</span>
      </code>
    </pre>

  

</body></html>"
`;
